packages/xstate-immer/src/index.ts

```diff
@@ -2,51 +2,62 @@ import {
   EventObject,
   ActionObject,
   AssignAction,
-  assign as xstateAssign
+  assign as xstateAssign,
+  AssignMeta
 } from 'xstate';
-import { produce, Draft, produceWithPatches, applyPatches } from 'immer';
+import { produce, Draft } from 'immer';
 
 export type ImmerAssigner<TContext, TEvent extends EventObject> = (
   context: Draft<TContext>,
-  event: TEvent
+  event: TEvent,
+  meta: AssignMeta<TContext, TEvent>
 ) => void;
 
 export interface ImmerAssignAction<TContext, TEvent extends EventObject>
   extends ActionObject<TContext, TEvent> {
   assignment: ImmerAssigner<TContext, TEvent>;
 }
 
-export function assign<TContext, TEvent extends EventObject = EventObject>(
-  assignment: ImmerAssigner<TContext, TEvent>
+function immerAssign<TContext, TEvent extends EventObject = EventObject>(
+  recipe: ImmerAssigner<TContext, TEvent>
 ): AssignAction<TContext, TEvent> {
-  // @ts-ignore (possibly infinite TS bug)
-  return xstateAssign((context, event) => {
-    return produce(context, (draft) => void assignment(draft, event));
+  return xstateAssign((context, event, meta) => {
```

Comment:
Awesome stuff!

NIT: Pretty sure this could be written as `return xStateAssign(produce(recipe))`, as `produce` curries and passes any additional arguments in the same order